transitclock/src/main/java/org/transitclock/avl/AvlClient.java

```diff
@@ -165,7 +167,7 @@ public void run() {
 				// filter
 				// the next one
 				avlReports.put(avlReport.getVehicleId(), avlReport);
-			}
+
```

Comment:
it the avlReports.put(...) is the problem here, I suggest extending the synchronized (avlReports) block instead. The Reentrant lock doesn't give you anything different as I understand it.